torch/csrc/inductor/aoti_eager/kernel_holder.h

```diff
@@ -14,9 +14,33 @@
 
 namespace torch::inductor {
 
-struct AOTIKernelState {
+// Represent AOTI kernel. It contains all the parameter metadata of the kernel
+// and the AOTI model runner.
+struct AOTIKernelMetadata {
+  // Represent all the parameters of AOTI kernel
+  std::vector<ParameterMetadata> parameter_metadata_list_;
+  // AOTI model runner to run the AOTI kernel
   std::shared_ptr<AOTIModelContainerRunner> kernel_runner_;
-  std::vector<torch::dynamo::TensorCheck> tensor_checks_;
+  AOTIKernelMetadata() : parameter_metadata_list_(), kernel_runner_(nullptr) {}
+
+  // Check whether the given parameter metadata list is the same as the
+  // parameter metadata list of the AOTI kernel.
+  bool check(
```

Comment:
`operator==`?

Reply:
It only checks its `parameter_metadata_list_` member and bypasses the `kernel_runer_` check. `operator==` may not exactly match the semantic.